base/sparse/linalg.jl

```diff
@@ -585,39 +585,41 @@ function normestinv{T}(A::SparseMatrixCSC{T}, t::Integer = 2)
         S_old = copy(S)
         for j = 1:t
             for i = 1:n
-                S[i,j] = Y[i,j]==0?1:sign(Y[i,j])
+                S[i,j] = Y[i,j]==0?1:(Y[i,j]/abs(Y[i,j]))
             end
         end
 
-        # Check wether cols of S are parallel to cols of S or S_old
-        for j = 1:t
-            done = false
-            while ~done
-                repeated = false
-                if j > 1
-                    saux = S[1:n,j]' * S[1:n,1:j-1]
-                    for i = 1:j-1
-                        if abs(saux[i]) == n
-                            repeated = true
-                            break
+        if T<:Real
+            # Check whether cols of S are parallel to cols of S or S_old
+            for j = 1:t
+                done = false
```

Comment:
Since this code was pre-existing, let's leave it as is in this PR and make tweaks to it separately. It would also be better to write `!done` and so on instead of the Matlabism `~done`.